### PR TITLE
Fixes #12360: It's difficult to know if the node property added is valid json 

### PR DESCRIPTION
**Pull request by RaphaelGauthier**
https://www.rudder-project.org/redmine/issues/12360

### CODE REVIEW

**Comment by RaphaelGauthier** [comment]
Commit modified

**Comment by RaphaelGauthier** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
What happens if you set a simple value ? I think it prevent from saving any value that is not a json

**Comment by VinceMacBuche** [comment]
Ok i understood it, I have some idea to make it clearer!!

File: rudder-web/src/main/webapp/javascript/rudder/angular/nodeProperties.js [hunk 1]
```diff
@@ -136,30 +138,35 @@ app.controller('nodePropertiesCtrl', function ($scope, $http, $compile) {
     }
     var propertyToSave = angular.copy($scope.newProperty)
     var newValue = propertyToSave.value
+    $scope.isJson = true;
     try {
       newValue = JSON.parse(propertyToSave.value)
     } catch(e) {
-      // Do nothing if error we keep the current value
+      //Check if Json validation is enabled. If so, the property is not saved and we display an error message, otherwise the current value is retained.
+      $scope.isJson = !$scope.checkJson
```
Comment (VinceMacBuche) [inline rudder-web/src/main/webapp/javascript/rudder/angular/nodeProperties.js]: and set "isValid" to false if you catch

File: rudder-web/src/main/webapp/templates-hidden/components/ComponentNodeProperties.html [hunk 2]
```diff
@@ -26,9 +26,18 @@
                   <td>
                     <span class="input-group-addon addon-json">=</span>
                   </td>
-                  <td class="form-group" ng-class="{'has-error': newPropForm.newPropValue.$error.required && newPropForm.newPropValue.$dirty}">
+                  <td class="form-group" ng-class="{'has-error': (newPropForm.newPropValue.$error.required && newPropForm.newPropValue.$dirty) || (checkJson && !isJson)}">
```
Comment (VinceMacBuche) [inline rudder-web/src/main/webapp/templates-hidden/components/ComponentNodeProperties.html]: Check should be on "isValid"

**Comment by RaphaelGauthier** [comment]
Commit modified

File: rudder-web/src/main/webapp/javascript/rudder/angular/nodeProperties.js [hunk 3]
```diff
@@ -136,30 +138,36 @@ app.controller('nodePropertiesCtrl', function ($scope, $http, $compile) {
...
-        $scope.properties.push(propertyToSave);
-        $scope.resetNewProperty();
-        $('#newPropPopup').bsModal('hide');
-        $scope.newPropForm.$setPristine();
+    $scope.isValid = true;
+    if($scope.checkJson){
+      try {
+        newValue = JSON.parse(propertyToSave.value)
+      } catch(e) {
+        //Check if Json validation is enabled. If so, the property is not saved and we display an error message, otherwise the current value is retained.
```
Comment (VinceMacBuche) [inline rudder-web/src/main/webapp/javascript/rudder/angular/nodeProperties.js]: You can update the comment (even remove it :) )

**Comment by RaphaelGauthier** [comment]
Commit modified

**Pull request merged**